istioctl/pkg/multixds/gather.go

```diff
@@ -152,22 +153,52 @@ func FirstRequestAndProcessXds(dr *xdsapi.DiscoveryRequest, centralOpts clioptio
 	return MultiRequestAndProcessXds(false, dr, centralOpts, istioNamespace, ns, serviceAccount, kubeClient)
 }
 
-func getXdsAddressFromWebhooks(client kube.ExtendedClient) (string, error) {
+type xdsAddr struct {
+	gcpProject, host, istiod string
+}
+
+func getXdsAddressFromWebhooks(client kube.ExtendedClient) (ret xdsAddr, _ error) {
 	webhooks, err := client.Kube().AdmissionregistrationV1().MutatingWebhookConfigurations().List(context.Background(), metav1.ListOptions{
 		LabelSelector: fmt.Sprintf("%s=%s,!istio.io/tag", label.IoIstioRev.Name, client.Revision()),
 	})
 	if err != nil {
-		return "", err
+		return ret, err
 	}
 	for _, whc := range webhooks.Items {
 		for _, wh := range whc.Webhooks {
 			if wh.ClientConfig.URL != nil {
 				u, _ := url.Parse(*wh.ClientConfig.URL)
-				return u.Host, nil
+				ret.host = u.Host
+				if strings.HasSuffix(u.Host, ".googleapis.com") {
```

Comment:
nit: break this out into a function in google.go

Reply:
Done